ActionScript/ActionScript.sublime-syntax

```diff
@@ -12,87 +12,892 @@ first_line_match: |-
+
+  comments:
+    - include: line-comments
+    - include: block-comments
+
+  block-comments:
+    - include: block-comment-basic
+
+  block-comment-basic:
+    - match: /\*
+      scope: punctuation.definition.comment.begin.actionscript
+      push: block-comment-basic-content
+
+  block-comment-basic-content:
+    - meta_include_prototype: false
+    - meta_scope: comment.block.actionscript
+
+    - match: \*/
+      scope: punctuation.definition.comment.end.actionscript
+      pop: 1
+
+    - match: ^\s*(\*)(?!/)
       captures:
-        1: keyword.declaration.class.actionscript.2
-        2: entity.name.type.class.actionscript.2
-        3: storage.modifier.extends.actionscript.2
-        4: entity.other.inherited-class.actionscript.2
+        1: punctuation.definition.comment.actionscript
+
+  line-comments:
+    - include: line-comment-double-slash
+
+  line-comment-double-slash:
+    - match: //
+      scope: punctuation.definition.comment.actionscript
+      push: line-comment-double-slash-content
+
+  line-comment-double-slash-content:
+    - meta_include_prototype: false
+    - meta_scope: comment.line.double-slash.actionscript
+
+    - include: eol-pop
+
+##[ Strings ]###########################################################################################################
+
+  strings:
+    - include: double-quoted-strings
+    - include: single-quoted-strings
+
+  double-quoted-strings:
+    - match: \"
+      scope: punctuation.definition.string.begin.actionscript
+      push: double-quoted-string-content
+
+  double-quoted-string-content:
+    - meta_include_prototype: false
+    - meta_scope: string.quoted.double.actionscript
+
+    - match: \"
+      scope: punctuation.definition.string.end.actionscript
+      pop: 1
+
+    - match: \\.
+      scope: constant.character.escape.actionscript
+
+  single-quoted-strings:
+    - match: \'
+      scope: punctuation.definition.string.begin.actionscript
+      push: single-quoted-string-content
+
+  single-quoted-string-content:
+    - meta_include_prototype: false
+    - meta_scope: string.quoted.single.actionscript
+
+    - match: \'
+      scope: punctuation.definition.string.end.actionscript
+      pop: 1
+
+    - match: \\.
+      scope: constant.character.escape.actionscript
+
+##[ Constants ]#########################################################################################################
+
+  constants:
+    - include: language-constants
+    - include: numeric-constants
+    - include: keyboard-input-constants
+
+  language-constants:
+    - match: \b{{constant_language}}\b
+      scope: constant.language.actionscript
+
+  numeric-constants:
+    - match: ([-+]??)(0[xX])(\h*)(L|l|UL|ul|u|U|F|f)?
+      scope: meta.number.integer.hexadecimal.actionscript
+      captures:
+        1: constant.numeric.sign.actionscript
+        2: constant.numeric.base.actionscript
+        3: constant.numeric.value.actionscript
+        4: constant.numeric.suffix.actionscript
```

Comment:
We probably don't want to include sign in numbers as it is not clear whether they can be distinguished from operators in all situations. The choosen `constant.numeric.sign` scope wouldn't be the appropriate one anyway.

```suggestion
    - match: (0[xX])(\h*)(L|l|UL|ul|u|U|F|f)?
      scope: meta.number.integer.hexadecimal.actionscript
      captures:
        1: constant.numeric.base.actionscript
        2: constant.numeric.value.actionscript
        3: constant.numeric.suffix.actionscript
```